modules/iterator/CMakeSources.cmake

```diff
@@ -0,0 +1,18 @@
+set(MODULE_PATH
+        ${CMAKE_CURRENT_SOURCE_DIR}/claws/iterator)
+
+set(MODULE_PUBLIC_HEADERS
+        "${MODULE_PATH}/circular_iterator.hpp"
+        "${MODULE_PATH}/iterator_pair.hpp"
```

Comment:
I think `claws::iterator_pair` could be a container actually, but I'm fine with this too